### PR TITLE
Check return codes in analyzejacobian.py

### LINKED ISSUE #9809: analyzejacobian.py doesn't check exit codes
### Description of the enhancement or error report

If an application fails for any reason, the analyzejacobian.py can fail to identify real problems with the Jacobian. That script really needs to check exit codes before trying to process output.

### Rationale for the enhancement or information for reproducing the error

Modify the anlyazejacobian.py script and add a bad CLI parameter (e.g. `Outputs/foobar=true`) where the command line is being assembled. The script will happily execute and parse the failed output and not detect any errors.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

Potential to report false negatives to users who are debugging their Jacobian statements.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #9809

### CODE REVIEW

**Comment by dschwen** [comment]
Actually we expect a return code of 1 in the FD phase... let me fix that

**Comment by dschwen** [comment]
So, since we expect a return code of 1 and that is exactly what any mooseError also produces I'm checking now if the matrices are parsed successfully. This seems to work quite well.

File: python/jacobiandebug/analyzejacobian.py [hunk 1]
```diff
@@ -397,10 +396,21 @@ def parseOutput(output, dofdata) :
     try:
         child = subprocess.Popen(mooseparams, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
         data = child.communicate()[0].decode("utf-8")
-        child.wait()
+        # analyze return code
+        if child.returncode == -11 :
+            print("The moose application crashed with a segmentation fault (try recompiling)")
+            sys.exit(1)
+        elif child.returncode != 1 :
```
Comment (rwcarlsen) [inline python/jacobiandebug/analyzejacobian.py]: Won't this include cases where ``returncode == 0``?

Comment (dschwen) [reply python/jacobiandebug/analyzejacobian.py]: Yes, it better. We expect a return code of 1 if the PETSc finite differencing worked.

Comment (permcody) [reply python/jacobiandebug/analyzejacobian.py]: Ah OK, That's good to know.

**Review by permcody — APPROVED** [review]

**Pull request merged**